code/modules/asset_cache/assets/vending.dm

```diff
@@ -30,10 +30,9 @@
 			continue
 
 		if (PERFORM_ALL_TESTS(focus_only/invalid_vending_machine_icon_states))
-			var/icon_states_list = icon_states(icon_file)
-			if (!(icon_state in icon_states_list))
+			if (!icon_exists(icon_file, icon_state))
 				var/icon_states_string
-				for (var/an_icon_state in icon_states_list)
+				for (var/an_icon_state in icon_states(icon_file))
```

Comment:
This isn't an optimization. Previous behaviour would store the icon states inside the `icon_states_list` and we would use that to both check if the icon state exist(in line 34) and if it did reuse that list again here.

Now `icon_exists()` proc will internally call `icon_states()` proc & we are again retrieving those same icon states in this for loop thus doing double the work